tst/EnergyPlus/unit/Fixtures/EnergyPlusFixture.hh

```diff
@@ -0,0 +1,194 @@
+		virtual void SetUp() {
+			show_message();
+			this->ioHelper = std::unique_ptr< IOHelper >( new IOHelper );
+		}
+
+		// This is run every unit test and makes sure to clear all state in global variables this fixture touches.
+		virtual void TearDown() {
+			CurveManager::clear_state();
+			DataEnvironment::clear_state();
+			DataGlobals::clear_state();
+			DataHeatBalance::clear_state();
+			DataIPShortCuts::clear_state();
+			DataOutputs::clear_state();
+			DataSurfaces::clear_state();
+			HeatBalanceManager::clear_state();
+			InputProcessor::clear_state();
+			OutputProcessor::clear_state();
+			ScheduleManager::clear_state();
+
+			{ 
+				IOFlags flags; 
+				flags.DISPOSE( "DELETE" );
+				gio::close( OutputProcessor::OutputFileMeterDetails, flags );
+				gio::close( DataGlobals::OutputFileStandard, flags ); 
+				gio::close( DataGlobals::OutputFileInits, flags );
+				gio::close( DataGlobals::OutputFileDebug, flags );
+				gio::close( DataGlobals::OutputFileZoneSizing, flags );
+				gio::close( DataGlobals::OutputFileSysSizing, flags );
+				gio::close( DataGlobals::OutputFileMeters, flags );
+				gio::close( DataGlobals::OutputFileBNDetails, flags );
+				gio::close( DataGlobals::OutputFileZonePulse, flags );
+
+			}
+		}
+
+		// This will output the "Begin Test" ShowMessage for every unit test that uses or inherits from this fixture.
+		// Now this does not need to be manually entered for every unit test as well as it will automatically be updated as the
+		// unit test names change.
+		void show_message() {
+			// Gets information about the currently running test.
+			// Do NOT delete the returned object - it's managed by the UnitTest class.
+			const ::testing::TestInfo* const test_info = ::testing::UnitTest::GetInstance()->current_test_info();
+			ShowMessage( "Begin Test: " + std::string( test_info->test_case_name() ) + ", " + std::string( test_info->name() ) );
+		}
+
+		static void use_cached_idd() {
+			IOHelper::use_cache();
+		}
+
+		// This will compare either a STL container or ObjexxFCL container
+		// Pass a forward iterator of a container you want to compare against an expected container. You can pass in an existing 
+		// container or use an initializer list like below.
+		// I had to only do one template typename due to wrapping this in Google Test macros causes the macro to fail. It thinks
+		// there are more macro arguments than expected. This is a way around that restriction.
+		// Example Usage: 
+		// 		EXPECT_TRUE( compare_containers< std::vector< bool > >( { true } , ObjectDef( index ).AlphaOrNumeric ) );
+		// 		EXPECT_TRUE( compare_containers< Array1D_bool >( { true } , ObjectDef( index ).AlphaOrNumeric ) );
+		template < typename T, class T2 >
+		bool compare_containers( T const & expected_container, T2 const & actual_container ) {
+			return ioHelper->compare_containers< T, T2 >( expected_container, actual_container );
```

Comment:
Yes ioHelper is private member variable. I pulled IOHelper into its own class to be able to unit test that new functionality easier. I originally had it all within EnergyPlusFixture but it was an ugly mess and not easily testable.

It can compare FCL to FCL, STL to FCL, and STL to STL. I could use std::equal to do this, yes, but I wanted to be able to check size of containers as well as output the values within the containers that differ and at what index.

There could definitely be something within E+ or FCL itself that just uses std::equal to compare any container types.